### PR TITLE
LogsignSiem Pack PR

### PR DESCRIPTION
**Pull request by keremvatandas**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [X] Yes
       - Further details:
   - [ ] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Review by Itay4 — COMMENTED** [review]
ill reach out to schedule a demo

**Comment by marcellarichmond** [comment]
@mchasepan fyi

File: Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      required: true
+      secret: false
+    deprecated: false
+    description: Query for columns
+    execution: false
+    name: logsign-get-columns-query
+    outputs:
+    - contextPath: LogsignSiem.Columns
+      description: Column list returned as the result of the query
+      type: Unknown
```
[the remark this answers is not in the document]
Comment (keremvatandas) [reply Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.yml:112]: Actually, this type is List<String>. Example output:  `"columns": ["127.0.0.1", "192.168.1.1", ...]`
I didn't find a species that supports this. That's why I wrote it as Unknown.

File: Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  type: 4
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+- additionalinfo: Last run time format like '%Y-%m-%dT%H:%M:%SZ'
```
[the remark this answers is not in the document]
Comment (keremvatandas) [reply Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.yml:24]: We using demisto getLastRun() and setLastRun() functions. If last_run is left blank (or None), we look at the first_fetch value. If first_fetch value is empty, '1 hour' is calculated as default value. (Now - first_fetch value) 
Users can enter values in the appropriate time format if they wish. (ISO8601 format with UTC)

File: Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
+category: Analytics & SIEM
+commonfields:
+  id: LogsignSiem
+  version: -1
+configuration:
```
[the remark this answers is not in the document]
Comment (keremvatandas) [reply Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.yml:5]: We have query support, but it is not required. In this version, we do not need to use active.

File: Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.py [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+    last_fetch = last_run.get('last_fetch', None)
+
+    if last_fetch is None:
+        last_fetch = datetime.utcnow() - timedelta(hours=int(first_fetch.split()[0]))
+    else:
+        last_fetch = datetime.strptime(last_fetch, DATE_FORMAT)
+
+    latest_created_time = last_fetch
+
+    data = client.get_incidents(method='GET', last_run=api_last_run, query=query)
```
[the remark this answers is not in the document]
Comment (keremvatandas) [reply Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.py]: I fixed it

**Review by Itay4 — COMMENTED** [review]
thanks for the demo
a few comments that came up:
1. please remove the `logsign-fetch-incidents` command or reimplement it
2. please. add the `Query` integration parameter
3. in the integration `Help` section, please keep only data on how to configure the instance (for example how to retrieve the api key)
4. please look into the failures in the `logsign-get-columns-query` command
5. please improve the pack readme as discussed according to https://xsoar.pan.dev/docs/documentation/pack-docs#pack-readme

File: Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.yml [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+  type: 0
+- display: Incident type
+  name: incidentType
+  required: false
+  type: 13
+- display: Fetch incidents
+  name: isFetch
+  required: false
+  type: 8
+- additionalinfo: You can use single quotes inside double quotes in query parameter.
```
Comment (Itay4) [inline Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem.yml:46]: consider adding default query here

File: Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem_description.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+This Content Pack is used to fetch incident logs, detail investigation on data lake, finding threats and anomaly to respond these threats in realtime.
+
+### Configure Logsign SIEM Integration on Cortex XSOAR
```
Comment (Itay4) [inline Packs/LogsignSiem/Integrations/LogsignSiem/LogsignSiem_description.md:3]: note my comment previous about this, this is not needed

File: Packs/LogsignSiem/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+`logsign-get-count-query`
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| query | Query. | Required | 
+| grouped_column | Grouped Column. | Required | 
+| criteria | Criteria Value. Possible values are: unique,  value. | Required | 
+| time_frame | Time Frame. | Required |
```
Comment (Itay4) [inline Packs/LogsignSiem/README.md:79]:
```suggestion
```

**Review by Itay4 — COMMENTED** [review]
the `logsign-get-incidents` command still exist as is, and as i've mentioned in the demo it currently has no value. please either remove it or reimplement as the other commands which populate to the context data

**Review by Itay4 — APPROVED** [review]

**Pull request merged**